src/styles/style-helpers.ts

```diff
@@ -25,6 +26,134 @@ function isTheme(args: ThemeArgs): args is Theme {
 export const getTheme = (args: ThemeArgs): Theme =>
   isTheme(args) ? args : args.theme;
 
+type Spacing = keyof Theme['spacings'] | null;
+
+const spacingString = (size: Spacing, theme: Theme) => {
+  if (!size) {
+    if (process.env.NODE_ENV !== 'production') {
+      console.warn(
+        'You are trying to reset the spacing which is not supported, because `null` means no margin, not zero margin. Fix it by providing one of the supported values.',
+      );
+    }
+    return null;
+  }
+  return css({ margin: theme.spacings[size] });
+};
+
+const spacingObject = (
+  size: { top?: Spacing; bottom?: Spacing; right?: Spacing; left?: Spacing },
+  theme: Theme,
+) => {
+  const margins: {
+    marginTop?: string;
+    marginBottom?: string;
+    marginRight?: string;
+    marginLeft?: string;
+  } = {};
+  if (size.top) {
+    margins.marginTop = theme.spacings[size.top];
+  }
+
+  if (size.bottom) {
+    margins.marginBottom = theme.spacings[size.bottom];
+  }
+
+  if (size.right) {
+    margins.marginRight = theme.spacings[size.right];
+  }
+
+  if (size.left) {
+    margins.marginLeft = theme.spacings[size.left];
+  }
+
+  return css(margins);
+};
+const spacingArray = (
+  size: [Spacing, Spacing?, Spacing?, Spacing?],
+  theme: Theme,
+) => {
+  const margins: {
+    marginTop?: string;
+    marginBottom?: string;
+    marginRight?: string;
+    marginLeft?: string;
+  } = {};
+
+  if (size.length === 1) {
+    if (!size[0]) {
+      if (process.env.NODE_ENV !== 'production') {
+        console.warn(
+          'You are trying to reset the spacing which is not supported, because `null` means no margin, not zero margin. Fix it by providing one of the supprted values',
```

Comment:
```suggestion
          'A single `null` value was passed to the spacing mixin. This has no effect since `null` represents no set margin, not zero margin. If you are trying to reset the spacing, use custom styles instead.',
```